src/Hox/Rendering.fs

```diff
@@ -215,58 +236,44 @@ module Builder =
   }
 
 /// This module contains functions that are used to render a node to a sequence of strings
+
 /// As soon as a chunk is ready it is yielded to the caller.
 [<RequireQualifiedAccess>]
 module Chunked =
 
   /// <summary>
   /// Renders the node and it's children to an asynchronous sequence of strings
-  /// The sequence is chunked by node and we keep track of the depth of the node
-  /// so we can choose between rendering recursively or buffering the async sequences
-  /// to avoid stack overflows.
   /// </summary>
-  /// <param name="stack">The stack of nodes to render</param>
-  /// <param name="cancellationToken">The cancellation token to use</param>
-  /// <returns>An asynchronous sequence of strings</returns>
-  /// <remarks>
-  /// This function will switch between rendering recursively and buffering the async sequences
-  /// depending on the depth of the node for nodes 200+ levels deep we'll default to buffering the chunk.
-  /// </remarks>
-  let rec renderNode
-    (
-      stack: Stack<struct (Node * bool * int)>,
-      cancellationToken: CancellationToken
-    ) : IAsyncEnumerable<string> =
-    taskSeq {
-      let cachedHtmlEncode =
-        getEncodedCache EscapeMode.Html (Dictionary<string, string>())
+  let renderNode (n : Node, cancellationToken : CancellationToken) : IAsyncEnumerable<string> =
+    let cachedHtmlEncode =
+      getEncodedCache EscapeMode.Html (Dictionary<string, string>())
 
-      let cachedAttrEncode =
-        getEncodedCache EscapeMode.Attribute (Dictionary<string, string>())
+    let cachedAttrEncode =
+      getEncodedCache EscapeMode.Attribute (Dictionary<string, string>())
 
-      while stack.Count > 0 do
+    taskSeq {
+      let dfs = Stack<Node * bool>()
```

Comment:
Converted to using a stack internally. Doesn't require knowing depth anymore either.